Changes for virtualization

Mark-Powers opened the pull request:
Not intended to merge, as this branch is incompatible with baremetal, but just to show a useful diff.

Main difference between baremetal is [99b59a6](https://github.com/ChameleonCloud/horizon/pull/28/commits/99b59a6201fba612df9641ce8391cf0bc2e463cc) which customizes the flavor selection panel of instance create To add description/gpu information.

mrhanlon committed:
Remove disk partition from rebuild instance form

Commit:
Hide stats that are irrelevant for baremetal

priteau committed:
Update instance actions to show only supported actions

Commit:
Add reservation dropdown to Angular launch-instance form

Commit:
Add reservation dropdown to Angular launch-instance form

Commit:
Port reservation changes for classic forms

Commit:
Hide AZ controls

xandroc committed:
Removing flavor details from instance launch panel

Flavor details are generic and don't reflect the actual state of the
hardware on Chameleon. Hide this panel from the user to avoid confusion.

Ticket: https://collab.tacc.utexas.edu/issues/11548

Commit:
Support sharing of images to Appliance catalog

This adds extra UI in the images pane to show which images are official
Chameleon images and also allows users to publish images so that they
can be shared by other users -- this is accomplished using the 'sharing'
property already supported by Glance.

Commit:
Support sharing of images to Appliance catalog

This adds extra UI in the images pane to show which images are official
Chameleon images and also allows users to publish images so that they
can be shared by other users -- this is accomplished using the 'sharing'
property already supported by Glance.

xandroc committed:
Displaying physical host running each instance

Commit:
Detect region when using versionless id endpoints

When using versionless Keystone endpoints in the service catalog (which
is the recommended way as it allows the client to do version selection),
region detection was broken. The reason is that a v3 suffix was
automatically added to the identity service URL before doing the region
detection. Region detection works by querying the service catalog for
which endpoint matches the Keystone auth URL used for the request. Since
the endpoint URL does not have a version suffix, the search will never
succeed, so the region defaults to None, and then later the first
endpoint is picked as a fallback behavior.

This fixes the check to also look for the original, unfixed URL when
trying to detect the region.

Commit:
Detect region when using versionless id endpoints

When using versionless Keystone endpoints in the service catalog (which
is the recommended way as it allows the client to do version selection),
region detection was broken. The reason is that a v3 suffix was
automatically added to the identity service URL before doing the region
detection. Region detection works by querying the service catalog for
which endpoint matches the Keystone auth URL used for the request. Since
the endpoint URL does not have a version suffix, the search will never
succeed, so the region defaults to None, and then later the first
endpoint is picked as a fallback behavior.

This fixes the check to also look for the original, unfixed URL when
trying to detect the region.

xandroc committed:
Using new portal endpoint & cache to fix slow loading images list

xandroc committed:
Using new portal endpoint & cache to fix slow loading images list

xandroc committed:
adding call to check for/fetch published appliances, generating & saving appliance dict

xandroc committed:
adding call to check for/fetch published appliances, generating & saving appliance dict

zhenz committed:
[swift] better error message when container name exists

zhenz committed:
[swift] better error message when container name exists

xandroc committed:
setting default flavor if only one option is available

xandroc committed:
setting default flavor if only one option is available

Commit:
Fix detection of regions for unversioned endpoints

When endpoints don't have Keystone versions in them, region detection in
Horizon fails. This is such a pain in the ass at this point I don't have
much else to say.

Commit:
Fix detection of regions for unversioned endpoints

When endpoints don't have Keystone versions in them, region detection in
Horizon fails. This is such a pain in the ass at this point I don't have
much else to say.

Commit:
Add toggle for region dropdown

This adds a toggle that allows an operator to disable the region
selector. This can be done by updating the configuration:

  OPENSTACK_KEYSTONE_MULTIREGION_SUPPORT = False

Commit:
Add toggle for region dropdown

This adds a toggle that allows an operator to disable the region
selector. This can be done by updating the configuration:

  OPENSTACK_KEYSTONE_MULTIREGION_SUPPORT = False

Commit:
Add multi-site support

This adds a new way to render the top navigation. Instead of showing the
regions as dropdowns, instead show a list of Chameleon sites, fetched
from the reference API. This allows us to further obscure the
multi-region capabilities of Chameleon, with the aim of ultimately
removing them.

To opt-in to this functionality an operator can define:

  CHAMELEON_MULTISITE_SUPPORT = True

Additionally, the site ID needs to be specified. This needs to map to an
ID of a site in the reference API:

  CHAMELEON_SITE_ID = "tacc"

Finally, to disable the existing multi-region dropdown:

  OPENSTACK_KEYSTONE_MULTIREGION_SUPPORT = False

This only affects whether the UI is shown and doesn't modify any other
functionality/capabilities.

Commit:
Add multi-site support

This adds a new way to render the top navigation. Instead of showing the
regions as dropdowns, instead show a list of Chameleon sites, fetched
from the reference API. This allows us to further obscure the
multi-region capabilities of Chameleon, with the aim of ultimately
removing them.

To opt-in to this functionality an operator can define:

  CHAMELEON_MULTISITE_SUPPORT = True

Additionally, the site ID needs to be specified. This needs to map to an
ID of a site in the reference API:

  CHAMELEON_SITE_ID = "tacc"

Finally, to disable the existing multi-region dropdown:

  OPENSTACK_KEYSTONE_MULTIREGION_SUPPORT = False

This only affects whether the UI is shown and doesn't modify any other
functionality/capabilities.

Commit:
Directly append page_url to site links

Commit:
Directly append page_url to site links

Commit:
Show projects by nickname in top dropdown

Commit:
Show projects by nickname in top dropdown

Commit:
Add authorized projects to context for clouds.yaml

This isn't used in the default template that ships in Horizon, but we
may make use of it in a custom template.

Commit:
Add authorized projects to context for clouds.yaml

This isn't used in the default template that ships in Horizon, but we
may make use of it in a custom template.

hlbfrancois committed:
Add /v3 to keystone endpoint url in app credential openrc template

Add /v3 to keystone endpoint url in app credential openrc template

hlbfrancois committed:
Add /v3 to keystone endpoint url in app credential openrc template

Add /v3 to keystone endpoint url in app credential openrc template

Commit:
Only render deep site links for same-class clouds

We broadly have two classes of clouds now, baremetal and edge. Deep
links do not translate between the two b/c they have different services
set up, often. To avoid too many 404s but also assist users in switching sites,
only render deeplinks if linking b/w sites of the same class.

Commit:
Only render deep site links for same-class clouds

We broadly have two classes of clouds now, baremetal and edge. Deep
links do not translate between the two b/c they have different services
set up, often. To avoid too many 404s but also assist users in switching sites,
only render deeplinks if linking b/w sites of the same class.

Commit:
Gracefully handle nova being disabled in fip view

If Nova is not in the deployment (really only applies for a Zun-only
deployment like CHI@Edge), the floatingip UI will fail because it
makes calls to Nova. Instead, we should just swallow that error.

Commit:
Gracefully handle nova being disabled in fip view

If Nova is not in the deployment (really only applies for a Zun-only
deployment like CHI@Edge), the floatingip UI will fail because it
makes calls to Nova. Instead, we should just swallow that error.

super-cooper committed:
Move uncommon tabs to advanced section (#22)

* Advanced tabs are always hidden when the workflow is launched
* Clicking the "Show Advanced" button does the following:
    * Changes the button to "Hide Advanced"
    * Makes advanced tabs appear
* Clicking the "Hide Advanced" advanced button does the following:
    * Changes the button to "Show Advanced"
    * Makes the advanced tabs disappear
* Using the "Next" or "Back" buttons to scroll to an advanced tab does the following:
    * Causes all the advanced tabs to appear
    * Changes the button to "Hide Advanced"
    * The advanced tabs will only disappear again if you click "Hide Advanced"
* If you hide advanced tabs while an advanced tab is selected:
    * The tab will remain selected
    * All the advanced tabs will be hidden on the sidebar

deshipu committed:
Respect SSL settings in placement API

Make the placement API client respect the OPENSTACK_SSL_NO_VERIFY and
OPENSTACK_SSL_CACERT configuration options, so that it can work properly
in TLS-everywhere deployments.

Change-Id: Id0bb085bdf411eef240c3d50da56016c0a1d075c
(cherry picked from commit 14212342cf8f7eb987e50de112958af31063e02e)

msherman64 committed:
update blazar-dashboard for angular 1.8.2

msherman64 committed:
update blazar-dashboard for angular 1.8.2

msherman64 committed:
fix quota display when only one quota is shown

When the displayed quotas are overridden to only show one chart,
(such as showing instances, but not vcpu/memory under compute)
the "row" html tag was not closed.

While this could be fixed by removing the check for "forloop.first" in
the closing bounds check, a more robust fix seems to be moving the row
open/close tags outside of the loop entirely.

Those checks were only in place to generate a new row every 6 columns,
but that is now handled by the bootstrap column spec instead, so we can
unconitionally generate a row for each section, and let bootstrap handle
rendering/wrapping of the columns.

msherman64 committed:
fix quota display when only one quota is shown

When the displayed quotas are overridden to only show one chart,
(such as showing instances, but not vcpu/memory under compute)
the "row" html tag was not closed.

While this could be fixed by removing the check for "forloop.first" in
the closing bounds check, a more robust fix seems to be moving the row
open/close tags outside of the loop entirely.

Those checks were only in place to generate a new row every 6 columns,
but that is now handled by the bootstrap column spec instead, so we can
unconitionally generate a row for each section, and let bootstrap handle
rendering/wrapping of the columns.

msherman64 committed:
instances: toggle disable secgroup and servergroup

make the security group and servergroup tab visibility toggle-able in
the instance launch dialog.

msherman64 committed:
fix error handling for default flavor

our logic to set the default flavor if only one is defined had the
potential to bypass the flavor validation logic. to resolve this, it was
moved outwards from the "build" to "update" method, and now checks the
"enabled" property, which is set after validation.

Additionally, logic was adjusted for the validation method, which would
raise errors when allocation flavor was set programatically, as the form
was not yet initialized.

msherman64 committed:
fix error handling for default flavor

our logic to set the default flavor if only one is defined had the
potential to bypass the flavor validation logic. to resolve this, it was
moved outwards from the "build" to "update" method, and now checks the
"enabled" property, which is set after validation.

Additionally, logic was adjusted for the validation method, which would
raise errors when allocation flavor was set programatically, as the form
was not yet initialized.

msherman64 committed:
allow configuring default flavor by name

in the instance launch dialog, allow configuration of default selected
flavor. To do so, set `default_flavor_name` in the `LAUNCH_INSTANCE_DEFAULTS`
configuration dict.

msherman64 committed:
allow configuring default flavor by name

in the instance launch dialog, allow configuration of default selected
flavor. To do so, set `default_flavor_name` in the `LAUNCH_INSTANCE_DEFAULTS`
configuration dict.

msherman64 committed:
BAREMETAL HACK: hide non-baremetal flavor info

the baremetal flavor doesn't make use of vcpus, ram, disk, and so on.
This is difficult to hide for ONLY a baremetal flavor, so instead just
rip the columns and charts out.

TODO: This will need to be reverted if VMs are deployed on the same
site as baremetal, or if we ever want to deploy this fork of horizon for
KVM.

A potential alternative is to apply this change via horizon override.
see: https://docs.openstack.org/horizon/latest/configuration/customizing.html#horizon-customization-module-overrides

msherman64 committed:
trivial: revert changes to whitespace/comments

msherman64 committed:
BAREMETAL: Disable server_group and secgroup panel

msherman64 committed:
instance-launch form: fix visibility of "advanced"

if a tab sequentiially "after" a tab marked "advanced" was selected,
all advanced tabs became visisble. Removing this code fixes that
behavior. Now, if an advanced tab is selected, then "show advanced" is
toggled off, the hidden tab will remain selected.

Since this matches the documented behavior in the commit that introduced
this, I propose that it's ok.

msherman64 committed:
Revert "[swift] better error message when container name exists"

This reverts commit 100e80a64ecc5c94a0898201ef64c8aad5d79eed.
Unsure why, but it breaks container creation in 2023.1+

msherman64 committed:
Revert "[swift] better error message when container name exists"

This reverts commit 100e80a64ecc5c94a0898201ef64c8aad5d79eed.
Unsure why, but it breaks container creation in 2023.1+

msherman64 committed:
Merge remote-tracking branch 'upstream/stable/2023.1' into chameleoncloud/2023.1

Mark-Powers committed:
Revert "BAREMETAL: Disable server_group and secgroup panel"

This reverts commit 08fb406535943999568642ad1fe57ba3dd6b8e3e.

Mark-Powers committed:
Revert "BAREMETAL: Disable server_group and secgroup panel"

This reverts commit 08fb406535943999568642ad1fe57ba3dd6b8e3e.

Mark-Powers committed:
Revert "BAREMETAL HACK: hide non-baremetal flavor info"

This reverts commit a63a8b22f63a3d02e790669a0e4a88a6a77610f7.

Mark-Powers committed:
Revert "BAREMETAL HACK: hide non-baremetal flavor info"

This reverts commit a63a8b22f63a3d02e790669a0e4a88a6a77610f7.

Mark-Powers committed:
Revert "Add reservation dropdown to Angular launch-instance form"

This reverts commit f4f3d277a0dd74fdb43128a11592543cfa88492b.

Mark-Powers committed:
Improve instance launch panel for flavor res use

msherman64 commented:
@Mark-Powers do you see an issue with, (for a while), maintaining this as a kvm-only branch?

As we still have not made the bare-metal only features toggle-able via config

although we'd probably have a smaller diff against chameleoncloud/2023.1, I'm unsure why some of the reverts are in this list when based against stable/2023.1

Mark-Powers commented:
@msherman64 Yes, I mentioned in the description, but we'll need to maintain this as it's own branch for now. I just made this PR to show the diff mainly

msherman64 commented on openstack_dashboard/api/nova.py:
```diff
@@ -258,10 +258,10 @@ def flavor_get(request, flavor_id, get_extras=False):
 @profiler.trace
-@memoized.memoized
-def flavor_list(request, is_public=True, get_extras=False):
+def flavor_list(request, is_public=True, get_extras=False, detailed=False):
```
flavor_list defaults to `detailed=True`, setting it to False here breaks default behavior for the instances table.

additionally, memoized is removed now?

msherman64 committed:
restore default behavior for flavor list

the default behavior for `flavor_list` is to include details. Setting the default to false breaks the table in project/instances.

pdmars committed:
Use chameleon-supported metadata on the image (#29)

* Use chameleon-supported metadata on the image
* Show Chameleon Support column in the image table with yes, no, deprecated
* Add feature flag for showing chameleon support column
* Add previous way to determine supported if not using the new metadata
* Fix previous behavior so it works correctly if the feature is disabled
* Address PR feedback: make the name of the chameleon-supported column configurable

Mark-Powers committed:
Filter out blazar reservation only flavors

These flavors should really only be seen by blazar, not the end user

Mark-Powers committed:
Fix horizon showing active reservation flavors

Mark-Powers committed:
Only show flavors from active reservations

pdmars committed:
Handle next parameter in redirect fed auth flow (#31)

pdmars committed:
Handle next parameter in redirect fed auth flow (#31) (#32)

Mark-Powers committed:
Add config for blazar flavor filter

Mark-Powers committed:
Merge branch 'chameleoncloud/2023.1' into chameleoncloud/2023.1-kvm

Mark-Powers committed:
Merge branch 'chameleoncloud/2023.1' into chameleoncloud/2023.1-kvm

Mark-Powers committed:
Add config for baremetal option

Mark-Powers commented:
Notes on latest commit 6a083914d533364383aa36ef2a3bd2a1edd2d50a

Takes all of our baremetal changes and puts them behind `CHAMELEON_BAREMETAL_ONLY` setting. This is propagated to the JS code via a wrapper for the built in horizon settings service. 

If set, this toggles everything to "baremetal mode", meaning we remove things that aren't functional/useful in baremetal. It's meant to be set only when we are using baremetal only. In the future when we have both, we will need a few more tweaks to the instance launch dialog to pick lease id or flavor.

This ended up being the 3rd iteration of the config, and it is the most reliable option I could find. The keypair tab already uses the settings API, so it shouldn't really slow anything down.

Mark-Powers committed:
Bring back upstream behavior for VMs
